Bug: #[server_fn] macro generates ::reqwest::Client requiring direct reqwest dependency in consumer crate

## Description

The `#[server_fn]` attribute macro in `reinhardt-pages` generates WASM client stubs that reference `::reqwest::Client::new()` directly. This requires the consumer crate to have `reqwest` as a direct dependency, even though `reinhardt-pages` already depends on `reqwest` for WASM targets.

## Steps to Reproduce

1. Create a crate that uses `#[server_fn]` from reinhardt-pages
2. Do NOT add `reqwest` to the crate's `Cargo.toml`
3. Build for `wasm32-unknown-unknown`

```
error[E0433]: failed to resolve: could not find `reqwest` in the list of imported crates
  --> src/apps/auth/server/login.rs:13:1
   |
13 | #[server_fn]
   | ^^^^^^^^^^^^ could not find `reqwest` in the list of imported crates
```

## Root Cause

In `crates/reinhardt-pages/macros/src/server_fn.rs`, the generated WASM stub uses:
```rust
let __client = ::reqwest::Client::new();
```

This resolves in the consumer crate's scope, where `reqwest` may not be a direct dependency.

## Expected Behavior

The macro should reference reqwest through the reinhardt-pages crate path, e.g.:
```rust
let __client = ::reinhardt::pages::__private::reqwest::Client::new();
```

Or reinhardt-pages should re-export reqwest so consumers can access it without a direct dependency.

## Workaround

Add `reqwest` as a direct WASM dependency in the consumer crate:
```toml
[target.'cfg(target_arch = "wasm32")'.dependencies]
reqwest = { version = "0.12", features = ["json"] }
```

## Environment

- reinhardt-web: main branch (2e1f5d76)
- Discovered during reinhardt-cloud WASM build setup (Issue kent8192/reinhardt-cloud#246)

--- a/crates/reinhardt-pages/macros/src/page/codegen.rs
+++ b/crates/reinhardt-pages/macros/src/page/codegen.rs
@@ -69,10 +69,13 @@ pub(super) fn generate(macro_ast: &TypedPageMacro) -> TokenStream {
 		body
 	};
 
-	// Wrap in a closure with conditional use statement if needed
+	// Wrap in a closure with conditional use statement if needed.
+	// #[allow(unused_variables)] suppresses warnings for closure parameters that are
+	// only used inside @event handlers, which are cfg-gated to wasm32 (#3327).
 	quote! {
 		{
 			#use_statement
+			#[allow(unused_variables)]
 			#params -> #pages_crate::component::Page {
 				#body_with_head
 			}

--- a/crates/reinhardt-pages/macros/src/server_fn.rs
+++ b/crates/reinhardt-pages/macros/src/server_fn.rs
@@ -638,7 +638,7 @@ fn generate_client_stub(
 			#serialize_code
 
 			// Build HTTP POST request with headers
-			let __client = ::reqwest::Client::new();
+			let __client = #pages_crate::__private::reqwest::Client::new();
 			let mut __request_builder = __client.post(&__endpoint)
 				.header("Content-Type", #content_type);
 

--- a/crates/reinhardt-pages/src/lib.rs
+++ b/crates/reinhardt-pages/src/lib.rs
@@ -249,5 +249,12 @@ pub use reinhardt_pages_macros::form;
 pub use reinhardt_pages_macros::head;
 pub use reinhardt_pages_macros::page;
 
+// Private re-exports used by macro-generated code. Not part of the public API.
+#[doc(hidden)]
+#[cfg(target_arch = "wasm32")]
+pub mod __private {
+	pub use reqwest;
+}
+
 // Logging macros are automatically exported via #[macro_export]
 // Users can access them as: reinhardt_pages::debug_log!, reinhardt_pages::info_log!, etc.

